lib/auth/grpcserver.go

```diff
@@ -354,13 +354,33 @@ func (g *GRPCServer) CreateAuditStream(stream authpb.AuthService_CreateAuditStre
 				switch {
 				case events.IsPermanentEmitError(err):
 					g.WithError(err).WithField("event", event).
-						Error("Failed to EmitAuditEvent due to a permanent error. Event wil be omitted.")
+						Error("Failed to RecordEvent due to a permanent error. Event wil be omitted.")
 					continue
 				default:
 					return trace.Wrap(err)
 				}
 			}
-			event.Size()
+
+			// Emit the event as well for v13 clients.
+			switch event.GetType() {
+			// Don't emit really verbose events.
+			case events.ResizeEvent, events.SessionDiskEvent, events.SessionPrintEvent, events.AppSessionRequestEvent, "":
+			default:
+				clientVersion, versionExists := metadata.ClientVersionFromContext(stream.Context())
+				if versionExists && semver.New(clientVersion).Major <= 13 {
+					if err := auth.EmitAuditEvent(stream.Context(), event); err != nil {
+						switch {
+						case events.IsPermanentEmitError(err):
+							g.WithError(err).WithField("event", event).
+								Error("Failed to EmitAuditEvent due to a permanent error. Event wil be omitted.")
+							continue
```

Comment:
should we return or continue at all?
Will it affect the session recording? Should we log and concurrently record the session to prevent the loss of remaining recordings?

Reply:
As it is now, returning the error here matches the behavior of v13. If the RPC exits before all session events have been recorded, the caller will just resume the audit stream and try again.